.pipelines/mac-ios-spm-extensions-validation-pipeline.yml

```diff
@@ -0,0 +1,101 @@
+        git checkout
+      workingDirectory: "$(Build.SourcesDirectory)"
+      displayName: "Sparse checkout objectivec/ folders from latest ORT main repository"
+    
+    - script: |
+        ls -R  "$(Build.SourcesDirectory)/tmp/onnxruntime"
+      workingDirectory: "$(Build.SourcesDirectory)/tmp"
+      displayName: "List sparse checkout repo contents"
+    
+    # Download artifacts for ORT C pod from iOS packaging pipeline
+    - task: DownloadPipelineArtifact@2
+      inputs:
+        buildType: 'specific'
+        project: 'Lotus'
+        definition: 995          #'definitionid' is obtained from `System.DefinitionId` of ORT CI: onnxruntime-ios-packaging-pipeline
+        buildVersionToDownload: 'latest'
+        branchName: 'main'
+        targetPath: '$(Build.ArtifactStagingDirectory)'
+
+    - script: |
+        set -e -x
+        ls
+      workingDirectory: '$(Build.ArtifactStagingDirectory)/$(artifactsName)'
+      displayName: "List staged artifacts for ORT C Pod"
+
+    # Download artifacts for ORT Ext C pod from extensions iOS packaging pipeline
+    - task: DownloadPipelineArtifact@2
+      inputs:
+        buildType: 'specific'
+        project: 'Lotus'
+        definition: 1206          #'definitionid' is obtained from `System.DefinitionId` of extensions CI: extensions.ios_packaging
+        buildVersionToDownload: 'specific'
+        branchName: 'yguo/change-header-path'   # TODO: update branch name here
+        pipelineID: 331190
+        targetPath: '$(Build.ArtifactStagingDirectory)'
+
+    - script: |
+        set -e -x
+        ls
+      workingDirectory: '$(Build.ArtifactStagingDirectory)/$(artifactsNameForExt)'
+      displayName: "List staged artifacts for ORT Ext C Pod"
+    
+    # Note: Running xcodebuild test on `onnxruntime-Package` scheme will perform swift unit tests for both OnnxRuntimeBindings
+    # and OnnxRuntimeExtensions targets.
+    - script: |
+        set -e -x
+        cd "$(Build.ArtifactStagingDirectory)/$(artifactsName)"
+        POD_ARCHIVE=$(find . -name "pod-archive-onnxruntime-c-*.zip")
+
+        shasum -a 256 "$(Build.ArtifactStagingDirectory)/$(artifactsName)/${POD_ARCHIVE}"
+        
+        cd "$(Build.ArtifactStagingDirectory)/$(artifactsNameForExt)"
+        POD_ARCHIVE_EXT=$(find . -name "pod-archive-onnxruntime-extensions-c-*.zip")
+
+        shasum -a 256 "$(Build.ArtifactStagingDirectory)/$(artifactsNameForExt)/${POD_ARCHIVE_EXT}"
+        
+        cd "$(Build.SourcesDirectory)/tmp/onnxruntime"
+        cp -r "$(Build.SourcesDirectory)/swift" .  
+        cp "$(Build.SourcesDirectory)/Package.swift" .
+        cp -r "$(Build.SourcesDirectory)/extensions" .
+
+        cp "$(Build.ArtifactStagingDirectory)/$(artifactsName)/${POD_ARCHIVE}" swift/
+        export ORT_IOS_POD_LOCAL_PATH="swift/${POD_ARCHIVE}"
+        cp "$(Build.ArtifactStagingDirectory)/$(artifactsNameForExt)/${POD_ARCHIVE_EXT}" swift/
+        export ORT_EXT_IOS_POD_LOCAL_PATH="swift/${POD_ARCHIVE_EXT}"
```

Comment:
nit: ORT_EXTENSIONS_IOS_POD_LOCAL_PATH would be slightly less full of 3 letter acronyms